python-connectors/googlesheets-sheet/connector.py

```diff
@@ -115,42 +122,27 @@ def __init__(self, config, parent, dataset_schema, dataset_partitioning, partiti
         self.dataset_schema = dataset_schema
         self.dataset_partitioning = dataset_partitioning
         self.partition_id = partition_id
-
         self.buffer = []
-
         columns = [col["name"] for col in dataset_schema["columns"]]
-
         if parent.result_format == 'first-row-header':
             self.buffer.append(columns)
 
-
     def write_row(self, row):
-
-        # Example of dataset_schema: {u'userModified': False, u'columns': [{u'timestampNoTzAsDate': False, u'type': u'string', u'name': u'condition', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'string', u'name': u'weather', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'double', u'name': u'temperature', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'bigint', u'name': u'humidity', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'date', u'name': u'date_update', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'date', u'name': u'date_add', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'string', u'name': u'ville', u'maxLength': -1}, {u'timestampNoTzAsDate': False, u'type': u'string', u'name': u'source', u'maxLength': -1}]}
-        # for (col, val) in zip(self.dataset_schema["columns"], row):
-        #     print (col, val)
-
         self.buffer.append(row)
 
-        # if len(self.buffer) > 50:
-        #     self.flush()
-
-
     def flush(self):
-        ws = get_spreadsheet(self.parent.credentials, self.parent.doc_id, self.parent.tab_id)
+        worksheet = self.parent.session.get_spreadsheet(self.parent.doc_id, self.parent.tab_id)
```

Comment:
It looks like when writing and no tab_id is given we just pick the first worksheet? It might be good to explicitly state that somewhere.

Reply:
If tab_id is not provided, an error is raised earlier, in `get_writer`

Reply:
I see - no problem then